src/averageAges.js

```diff
@@ -15,6 +15,20 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
+  const arrayOfMan = people.filter(({ sex }) => sex === 'm');
+
+  const arrayOfManCentury
+    = arrayOfMan.filter(({ died }) => Math.ceil(died / 100) === century);
+
+  const arrayOfManAge = arrayOfMan.map(({ died, born }) => died - born);
+  const arrayOfManAgeCentury
+    = arrayOfManCentury.map(({ died, born }) => died - born);
```

Comment:
You are performing lots of unnecessary opertaions.
You can first use `filter` to get all men and then, if `century` is passed, use another filter.

It will also allow you to have more simple averageAge calculation.
This comment is valid for all three tasks.